### PR TITLE
Bump engine and rds cert

### PR DESCRIPTION
**Pull request by samcrang**
Merge #371 first (and rebase).

### CODE REVIEW

**Comment by jakubmiarka** [comment]
The previous #371 is failing build now:
`Error: "ca_cert_identifier": this field cannot be set`
https://deployer.tools.signin.service.gov.uk/teams/main/pipelines/verify-self-service/jobs/deploy-to-staging/builds/487#L5e29bfb9:55

Do we need to update the provider first?

**Comment by samcrang** [comment]
@jakubmiarka Yup—I've fixed that now so this should be good to go.

**Comment by jakubmiarka** [comment]
> @jakubmiarka Yup—I've fixed that now so this should be good to go.

We need to go higher than the version you updated it too, because of this: https://github.com/terraform-providers/terraform-provider-aws/pull/10890

Until now, we've been using a fork: https://github.com/alphagov/verify-terraform-provider-aws

File: terraform/modules/self-service/db.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 resource "aws_db_instance" "self_service" {
   name           = "selfservice"
   engine         = "postgres"
-  engine_version = "10.6"
+  engine_version = "10.10"
```
Comment (jakubmiarka) [inline terraform/modules/self-service/db.tf]:
```suggestion
  engine_version = "11.5"
```
https://deployer.tools.signin.service.gov.uk/teams/main/pipelines/verify-self-service/jobs/deploy-to-production/builds/175#L5e29c042:146

Comment (samcrang) [reply terraform/modules/self-service/db.tf]: Hmmm. I wonder why production is running a major version newer version of Postgres compared to staging. I guess we can just do this change but it seem odd...

Comment (jakubmiarka) [reply terraform/modules/self-service/db.tf]: they were provisioned at different times (and the default version was different?).. would be my guess 🤷‍♂

**Pull request merged**